### PR TITLE
Enable WebAPI interface to handle parameters through constructor

### PR DESCRIPTION
**Pull request by phoenix128**
Add construct parameters injection for WebAPI interfaces.

### Description
WebAPI interfaces were relying on setter/getter methods violating the immutable stateless pattern.
This Pull Request enables the ServiceInputProcessor to get arguments directly from constructor. Setters methods are not required anymore.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: lib/internal/Magento/Framework/Webapi/ServiceInputProcessor.php [hunk 1]
```diff
@@ -82,6 +89,7 @@ public function __construct(
         AttributeValueFactory $attributeValueFactory,
         CustomAttributeTypeLocatorInterface $customAttributeTypeLocator,
         MethodsMap $methodsMap,
+        \Magento\Framework\ObjectManager\ConfigInterface $config,
```
Comment (VladimirZaets) [inline lib/internal/Magento/Framework/Webapi/ServiceInputProcessor.php]: Hi, @phoenix128 , due to [Magento backward-compatible guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) we can't add required parament to the constructor. 
Please add the optional parameter and take instance from `ObjectManager`, as example:
`$this->storeManager = $storeManager ?: \Magento\Framework\App\ObjectManager::getInstance()>get(\Magento\Store\Model\StoreManagerInterface::class);`

**Comment by VladimirZaets** [comment]
Hi @phoenix128  , any updates?

**Comment by phoenix128** [comment]
Hi @VladimirZaets , sorry if I did not answer you before. I've been terribly busy. You are definitely right about that, I must have forgotten it while coding quickly. Fixing in the next commit.

**Comment by VladimirZaets** [comment]
Hi @phoenix128. Thanks.
Also, optional param should be added to end of the params list. I fixed it.

**Comment by magento-engcom-team** [comment]
Hi @phoenix128. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.3.0 release.

**Comment by devamitbera** [comment]
@phoenix128

Can you please give an example how we can add **Add construct parameters injection for WebAPI interface**

**Pull request merged**